src/accelerate/accelerator.py

```diff
@@ -573,7 +558,7 @@ def main_process_first(self):
         ...     print(f"This will be printed by process {accelerator.process_index}")
         ```
         """
-        yield from self._goes_first(self.is_main_process)
+        yield self.state.main_process_first()
```

Comment:
Shoudn't this be `yield from`?

Reply:
Using `yield from` actually will raise an error and using `yield` with the inner calling `yield from` is enough. I'll add a test with this first but the multi-GPU tests passed :)